### PR TITLE
Mention caching POST with body in cache key

### PR DESCRIPTION
**Pull request by nibinqtl**

### CODE REVIEW

File: en/docs/edge-logic/faq.md [hunk 1]
```diff
@@ -128,7 +142,7 @@ If you only have the origin server in a central data center or in the cloud, the
...
 * **Cache the response!**
 
-In many cases, "dynamic" does not mean the content is not cacheable. For example, end users will not experiance the difference if you cache the score of a basketball game for 1 second. If you have 10 requests per second to fetch the score, you can save 90% of the bandwidth and processing power. If the response depends on some query parameter or request header values, make sure those variables are [included in the cache key](#how-do-you-include-query-parameters-andor-request-headers-in-the-cache-key).
+In many cases, "dynamic" does not mean the content is not cacheable. For example, end users will not experiance the difference if you cache the score of a basketball game for 1 second. If you have 10 requests per second to fetch the score, you can save 90% of the bandwidth and processing power. If the response depends on some query parameter, request header value or the request body, make sure all those variables are [included in the cache key](#how-do-i-include-query-parameters-request-headers-and-body-in-the-cache-key).
```
Comment (burtonquantil) [inline en/docs/edge-logic/faq.md:145]: Suggestion:
For example, end users will typically not notice if you cache the score of a basketball game for 1 second.

File: en/docs/recipes/api-acceleration.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ that the API user name and client IP are added to the cache key. This ensures th
 cached content will be served only to the same user from the same IP address. Combined
 with HTTPS and the short cache time of 1 minute, this should be reasonably safe for most 
 applications in the industry. By default, only responses to `GET` method are cached. 
-You can use the [proxy_cache_methods](/docs/edge-logic/supported-directives#proxy_cache_methods) directive to cache other responses.
+You can use the [proxy_cache_methods](/docs/edge-logic/supported-directives#proxy_cache_methods) directive to cache other responses. Refer to the [FAQ for an example](/docs/edge-logic/faq#how-do-i-include-query-parameters-request-headers-and-body-in-the-cache-key) to cache `POST` requests with request body in the cache key.
```
Comment (burtonquantil) [inline en/docs/recipes/api-acceleration.md:55]: Suggestion:  add the word "the" before  "request body".

In other words, change the sentence to end with
.... requests with the request body in the cache key.

**Review by burtonquantil — CHANGES_REQUESTED** [review]
I suggested a few changes to improve the text.

File: zh/docs/recipes/api-acceleration.md [hunk 3]
```diff
@@ -52,7 +52,7 @@ that the API user name and client IP are added to the cache key. This ensures th
 cached content will be served only to the same user from the same IP address. Combined
 with HTTPS and the short cache time of 1 minute, this should be reasonably safe for most 
 applications in the industry. By default, only responses to `GET` method are cached. 
-You can use the [proxy_cache_methods](/docs/edge-logic/supported-directives#proxy_cache_methods) directive to cache other responses.
+You can use the [proxy_cache_methods](/docs/edge-logic/supported-directives#proxy_cache_methods) directive to cache other responses. Refer to the [FAQ for an example](/docs/edge-logic/faq#如何将问号后参数，请求头，或者请求正文加入到缓存key中) to cache `POST` requests with the request body in the cache key.
```
Comment (Jianghai-Quantil) [inline zh/docs/recipes/api-acceleration.md:55]: I suggest changing the link to:
Refer to [this FAQ] (/docs/edge-logic/faq#如何将问号后参数，请求头，或者请求正文加入到缓存key中) for an example that explains how to cache `POST` requests with the request body in the cache key.

File: en/docs/recipes/api-acceleration.md [hunk 4]
```diff
@@ -52,7 +52,7 @@ that the API user name and client IP are added to the cache key. This ensures th
 cached content will be served only to the same user from the same IP address. Combined
 with HTTPS and the short cache time of 1 minute, this should be reasonably safe for most 
 applications in the industry. By default, only responses to `GET` method are cached. 
-You can use the [proxy_cache_methods](/docs/edge-logic/supported-directives#proxy_cache_methods) directive to cache other responses.
+You can use the [proxy_cache_methods](/docs/edge-logic/supported-directives#proxy_cache_methods) directive to cache other responses. Refer to the [FAQ for an example](/docs/edge-logic/faq#how-do-i-include-query-parameters-request-headers-and-body-in-the-cache-key) to cache `POST` requests with the request body in the cache key.
```
Comment (Jianghai-Quantil) [inline en/docs/recipes/api-acceleration.md:55]: I suggest changing the link to:
Refer to [this FAQ] (/docs/edge-logic/faq#how-do-i-include-query-parameters-request-headers-and-body-in-the-cache-key) for an example that explains how to cache `POST` requests with the request body in the cache key.

File: en/docs/recipes/api-acceleration.md [hunk 5]
```diff
@@ -52,7 +52,7 @@ that the API user name and client IP are added to the cache key. This ensures th
 cached content will be served only to the same user from the same IP address. Combined
 with HTTPS and the short cache time of 1 minute, this should be reasonably safe for most 
 applications in the industry. By default, only responses to `GET` method are cached.
```
Comment (Jianghai-Quantil) [inline en/docs/recipes/api-acceleration.md:54]: We actually cache GET and HEAD methods by default.

File: zh/docs/edge-logic/faq.md [hunk 6]
```diff
@@ -56,7 +56,21 @@ set $cache_misc "${cache_misc}hdr1=$http_header1&hdr2=$http_header2";
...
+最后给您的提醒是，不要忘了使用 [`proxy_cache_methods`](/docs/edge-logic/supported-directives.md#proxy_cache_methods) 指令来开启对 POST 请求的缓存。
+下面是如何把 POST 请求正文加入缓存 key 的配置代码示例：
+```nginx
+location /api/v1/ {
+  origin_pass my-api-origin;
+  proxy_cache_methods GET HEAD POST; # 允许缓存 POST 请求
+  proxy_cache_valid 1m; # 200, 301, 和 302 响应会被缓存1分钟
+  proxy_cache_bypass $ignored_body_in_cache_key;
+  proxy_no_cache $ignored_body_in_cache_key;
+  proxy_request_buffering on; # 必须打开此配置 proxy_request_body_in_cache_key 才能工作
```
Comment (Jianghai-Quantil) [inline zh/docs/edge-logic/faq.md:68]: I suggest changing the text to:
必须开启该配置，proxy_request_body_in_cache_key才能生效

File: zh/docs/edge-logic/faq.md [hunk 7]
```diff
@@ -131,7 +145,7 @@ proxy_ignore_headers Vary;
 当您使用 CDN Pro 时，您的客户端将会被全局调度系统（GSLB）引导与最近的边缘服务器建连。客户端与边缘服务器之间的往返时间 (RTT) 可能比客户端直连源服务器快几百毫秒。 TCP 和 TLS 握手通常需要 3-4 个 RTT，这样便可以通过 CDN 来提升1秒的响应性能。默认情况下，CDN Pro 与源站之间会保持长链接，您可以使用指令 [keep-alive timeout](/docs/portal/edge-configurations/managing-origins) 来设置长达10分钟的长链接时间。同时如果您已提前规划了某些业务不需要缓存，那么您可以使用指令 [`proxy_no_cache 1;`](</docs/edge-logic/supported-directives.md#proxy_no_cache>) 和 [`proxy_cache_bypass 1;`](</docs/edge-logic/supported-directives.md#proxy_no_cache>) 来跳过缓存处理步骤以最大程度减少 CDN Pro 上的延迟。
 * **将动态文件转换成可缓存文件**
 
-在许多情况下，“动态文件”并不意味着内容完全不可缓存。例如，如果您将篮球比赛的得分缓存 1 秒，客户端几乎体验不到任何差异。如果每秒有 10 个请求来获取分数，则可以节省 90% 的源站带宽和CDN执行损耗。需要注意的是，如果客户端收到的响应需要根据请求 URL 中的问号后参数或者请求头部值而不同的话，请确保[相关参数或者请求头已被添加到缓存 key 中](#如何将问号后参数或者请求头加入到缓存key中)。
+在许多情况下，“动态文件”并不意味着内容完全不可缓存。例如，如果您将篮球比赛的得分缓存 1 秒，客户端几乎体验不到任何差异。如果每秒有 10 个请求来获取分数，则可以节省 90% 的源站带宽和CDN执行损耗。需要注意的是，如果客户端收到的响应依赖请求 URL 中的参数或者请求头部值，或者请求正文的话，请确保所有相关参数都被[添加到缓存 key 中](#如何将问号后参数，请求头，或者请求正文加入到缓存key中)。
```
Comment (Jianghai-Quantil) [inline zh/docs/edge-logic/faq.md:148]: “CDN执行损耗” 改成 “算力”？ 这里节省的算力，也是源站的，不是CDN吧？

File: zh/docs/edge-logic/supported-directives.md [hunk 8]
```diff
@@ -417,7 +417,7 @@ else { ... }
 **Default:** `keepalive_timeout 30s;`<br/>
 **Context:** server (仅限在LB7)
 
-第一个参数设置每个 keep-alive 连接的最长空闲时间。服务器会关闭空闲过长的连接。设置为 0 将会禁用 keep-alive 连接。第二个参数（非必填）用于设置 “Keep-Alive: timeout=time” 这个响应头里的值。这两个参数的数值可以不同，但皆不能超过300s。
+第一个参数设置每个与客户端 keep-alive 连接的最长空闲时间。服务器会关闭空闲过长的连接。设置为 0 将会禁用 keep-alive 连接。第二个参数（非必填）用于设置 “Keep-Alive: timeout=time” 这个响应头里的值。这两个参数的数值可以不同，但皆不能超过300s。
```
Comment (Jianghai-Quantil) [inline zh/docs/edge-logic/supported-directives.md:420]: I suggest changing the text to:
第一个参数用来设置CDN服务端与客户端 keep-alive 长连接的空闲超时时长。超过超时时长后，CDN服务端会断开连接。设置为 0 表示禁用 keep-alive 长连接。第二个参数（非必填）用于设置 “Keep-Alive: timeout=time” 这个响应头里的值。这两个参数的数值可以不同，但都不能超过300s。

**Review by Jianghai-Quantil — CHANGES_REQUESTED** [review]
Suggested some changes

File: zh/docs/recipes/api-acceleration.md [hunk 9]
```diff
@@ -9,7 +9,7 @@ duplicate API calls in a short period of time. Some caching using a CDN can sign
 improve the performance. In this article, we are going to use an example to illustrate how
 to use the portal to create a property to accelerate an API server.
 
-Here are the assumptions about this task:
+下面是本次假想任务的一些背景和需求:
```
Comment (Jianghai-Quantil) [inline zh/docs/recipes/api-acceleration.md:12]: I suggest changing the text to:
假设需要加速的API服务，其基本信息如下：

**Comment by Jianghai-Quantil** [comment]
I pushed this commit, because I noticed that the current image file in production is broken. 
https://github.com/mileweb/docs-content-cdn/pull/619/commits/0f2d870d671c9f03d2426baa9c6ee797f3d67914

File: en/docs/recipes/api-acceleration.md (hunk 5 above)
Comment (nibinqtl) [reply en/docs/recipes/api-acceleration.md:54]: Updated

File: zh/docs/recipes/api-acceleration.md (hunk 9 above)
Comment (nibinqtl) [reply zh/docs/recipes/api-acceleration.md:12]: 这个我更喜欢我的文风。

**Review by burtonquantil — APPROVED** [review]
The English updates look OK to me

**Review by Jianghai-Quantil — APPROVED** [review]
OK to merge.

**Pull request merged**